protocols/v2/noise-sv2/src/signature_message.rs

```diff
@@ -73,3 +93,119 @@ impl SignatureNoiseMessage {
+        let static_pk = XOnlyPublicKey::from_keypair(&kp);
+        let authority_kp = Keypair::new(&secp, &mut rng);
+        let authority_pk = XOnlyPublicKey::from_keypair(&authority_kp);
+
+        let mut msg: [u8; 74] = [0; 74];
+        msg[0..2].copy_from_slice(&1u16.to_le_bytes()); // version = 1
+        msg[2..6].copy_from_slice(&100u32.to_le_bytes()); // valid_from = 100
+        msg[6..10].copy_from_slice(&200u32.to_le_bytes()); // not_valid_after = 200
+
+        SignatureNoiseMessage::sign(&mut msg, &static_pk.0, &authority_kp);
+
+        let signed_message = SignatureNoiseMessage::from(msg);
+        let is_valid = signed_message.verify(&static_pk.0, &Some(authority_pk.0));
+
+        assert!(!is_valid);
+
+        let signed_message = SignatureNoiseMessage::from(msg);
+        let wrong_kp = Keypair::new(&secp, &mut rng);
+        let wrong_pk = XOnlyPublicKey::from_keypair(&wrong_kp);
+        let is_invalid = signed_message.verify(&static_pk.0, &Some(wrong_pk.0));
+
+        assert!(!is_invalid);
+    }
+
+    #[test]
+    fn test_verify_without_authority() {
+        let signed_message = SignatureNoiseMessage {
+            version: 1,
+            valid_from: 100,
+            not_valid_after: 200,
+            signature: [0; 64],
+        };
+
+        let is_valid = signed_message.verify(&XOnlyPublicKey::from_slice(&[1; 32]).unwrap(), &None);
+        assert!(is_valid);
+    }
+
+    #[test]
+    fn test_verify_outside_time_window() {
+        let secp = Secp256k1::new();
+        let mut rng = rand::thread_rng();
+        let kp = Keypair::new(&secp, &mut rng);
+        let static_pk = XOnlyPublicKey::from_keypair(&kp);
+        let authority_kp = Keypair::new(&secp, &mut rng);
+        let authority_pk = XOnlyPublicKey::from_keypair(&authority_kp);
+
+        let mut msg: [u8; 74] = [0; 74];
+        msg[0..2].copy_from_slice(&1u16.to_le_bytes()); // version = 1
+        msg[2..6].copy_from_slice(&100u32.to_le_bytes()); // valid_from = 100
+        msg[6..10].copy_from_slice(&110u32.to_le_bytes()); // not_valid_after = 110 (already expired)
+
+        SignatureNoiseMessage::sign(&mut msg, &static_pk.0, &authority_kp);
+
+        let signed_message = SignatureNoiseMessage::from(msg);
+        let is_valid = signed_message.verify(&static_pk.0, &Some(authority_pk.0));
+
+        assert!(
+            !is_valid,
+            "The signed message should be invalid if it is outside the valid time window."
+        );
+    }
+
+    #[test]
+    fn test_split_method() {
+        let message = SignatureNoiseMessage {
+            version: 1,
+            valid_from: 100,
+            not_valid_after: 200,
+            signature: [0; 64],
+        };
+
+        let (header, signature) = message.split();
+
+        assert_eq!(header[0..2], 1u16.to_le_bytes());
+        assert_eq!(header[2..6], 100u32.to_le_bytes());
+        assert_eq!(header[6..10], 200u32.to_le_bytes());
+
+        assert_eq!(signature, [0; 64]);
+    }
+}
```

Comment:
Remove these tests and list them in #1130.

```suggestion
```

Reply:
is there something wrong with these tests?

Reply:
Not really, she wants to have it as a doc specific PR, with no code additions.

Reply:
ah I see, I didn't see the original comment

yeah it would still be good to add these tests on a future PR addressing #1130

so I would keep them in a different branch and write a note to #1130 for future reference

Reply:
Documented here: https://github.com/stratum-mining/stratum/issues/1130